src/Kafka/ManageKafkaPermissions/components/SelectAccount.tsx

```diff
@@ -0,0 +1,187 @@
+                principal.principalType === PrincipalType.ServiceAccount
+            )
+            .sort((a, b) =>
+              a.displayName && b.displayName
+                ? a.displayName.localeCompare(b.displayName)
+                : -1
+            )
+            .map((principal, index) => (
+              <SelectOption
+                key={index}
+                value={principal.id}
+                description={principal.displayName}
+              >
+                {principal.id}
+              </SelectOption>
+            ))}
+        </SelectGroup>,
+        <Divider key="user_account_divider" />,
+        <SelectGroup
+          label={t("manage_permissions_dialog.all_accounts_user_account_group")}
+          key="user_accounts_group"
+        >
+          {initialOptions
+            .filter(
+              (principal) =>
+                principal.principalType === PrincipalType.UserAccount
+            )
+            .map((principal, index) => (
+              <SelectOption
+                key={index}
+                value={principal}
+                description={principal.displayName}
+              >
+                {principal.id}
+              </SelectOption>
+            ))}
+        </SelectGroup>,
+      ]
+    : [];
+
+  const customFilter = (
+    _: React.ChangeEvent<HTMLInputElement> | null,
+    value: string
+  ) => {
+    if (!value) {
+      return options;
+    }
+
+    const input = new RegExp(value, "i");
+    return options
+      .filter((accounts) => Array.isArray(accounts.props.children))
+      .map((account) =>
+        account.props.children.filter(
+          (allAccounts: { props: { value: string; description: string } }) =>
+            input.test(allAccounts.props.value) ||
+            input.test(allAccounts.props.description)
+        )
+      );
+  };
+
+  return (
+    <FormGroupWithPopover
+      labelHead={t("manage_permissions_dialog.account_id_title")}
+      fieldId="kafka-instance-name"
+      fieldLabel={t("manage_permissions_dialog.account_id_title")}
+      labelBody={t("manage_permissions_dialog.account_id_help")}
+      buttonAriaLabel={t("manage_permissions_dialog.account_id_aria")}
+      isRequired={true}
+      helperTextInvalid={id.errorMessage}
+      validated={id.validated || ValidatedOptions.default}
+    >
+      <Select
+        data-testid="acls-select-account"
+        variant={SelectVariant.typeahead}
+        className="kafka-ui--select--limit-height"
+        typeAheadAriaLabel={t(
+          "manage_permissions_dialog.account_id_typeahead_placeholder"
+        )}
+        onToggle={onToggle}
+        onSelect={onSelect}
+        onClear={clearSelection}
+        selections={id.value}
+        onFilter={customFilter}
+        isOpen={isOpen}
+        placeholderText={t(
+          "manage_permissions_dialog.account_id_typeahead_placeholder"
+        )}
+        placeholder={t(
+          "manage_permissions_dialog.account_id_typeahead_placeholder"
+        )}
```

Comment:
When I look at the [PF example for typeahead select](https://www.patternfly.org/v4/components/select#typeahead), I see a `placeholderText` prop, but not a `placeholder` prop. This is what provides the "Select an account" that displays inside the typeahead input. It would be good to check if we even need the `placeholder` here and remove it if not needed.